fix: [M3-6442] - Inefficient firewall custom ports regex

abailly-akamai opened the pull request:
## Description 📝

Fix a [codeQL inefficient regex warning](https://github.com/linode/manager/security/code-scanning/25) for firewall custom ports (exponential backtracking on strings containing many repetitions of '-') in the firewall rule drawer.

I increased the regex matching specificity to only match:
`42`
`1-10`
`1,2,3,4,5`

and disallow things like:
`1--10`
`1-2-3`
`1,2,3-4-5`
`1-2,3-4`

along with extra spaces etc

## Preview 📷
![Screenshot 2023-04-17 at 1 54 39 PM](https://user-images.githubusercontent.com/130582365/232570719-fbabc2b6-d0cb-463a-a6c2-5814c76fe031.jpg)

## How to test 🧪

- pull code locally, and add some rules at `/firewalls/[id]/rules`
- `yarn test FirewallRuleDrawer`
or
- test regex at https://regex101.com/ or other online tool

bnussman-akamai commented:
With the current regex, I can't create a firewall with the port to a comma separated list (`1,2,3,4,5`)

abailly-akamai commented:
@bnussman-akamai ah yeah good catch, we are formatting the input so the regex had to allow for whitespace. Updated it and added tests to cover for those cases for good measure:

- `42`
- `1-10`
- `1,2,3,4,5`
- `1, 2, 3, 4, 5`

abailly-akamai commented:
@hkhalil-akamai IMO It's better to keep that validation only on the back-end (shall it ever change) and it's not a formatting validation which this PR is meant to target
